From 5f4cd47a2f8c85e43549a324729347201df493b4 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Fri, 13 Jun 2014 10:30:57 +0100
Subject: [PATCH 227/351] switch: arizona: Update hpdet_cb to include mic
 status

It is sometime useful to know whether a microphone is connected in the
machine driver. This patch folds this information in the headphone
detection callback, as this will already alert the machine driver on
each successful jack detection.

Change-Id: I59976b8f10d1863f7c032bcb6ea0f84a00c8c7c6
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c   |    4 ++--
 include/linux/mfd/arizona/pdata.h |    2 +-
 sound/soc/codecs/arizona.c        |    2 +-
 sound/soc/codecs/arizona.h        |    3 ++-
 4 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 372a102..3ddaed3 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -831,7 +831,7 @@ int arizona_hpdet_reading(struct arizona_extcon_info *info, int val)
 	arizona->hp_impedance = val;
 
 	if (arizona->pdata.hpdet_cb)
-		arizona->pdata.hpdet_cb(arizona->hp_impedance);
+		arizona->pdata.hpdet_cb(arizona->hp_impedance, info->mic);
 
 	switch (arizona->type) {
 	case WM5110:
@@ -1537,7 +1537,7 @@ static irqreturn_t arizona_jackdet(int irq, void *data)
 
 		/* Use a sufficiently large number to indicate open circuit */
 		if (arizona->pdata.hpdet_cb) {
-			arizona->pdata.hpdet_cb(ARIZONA_HP_Z_OPEN);
+			arizona->pdata.hpdet_cb(ARIZONA_HP_Z_OPEN, false);
 		}
 	}
 
diff --git a/include/linux/mfd/arizona/pdata.h b/include/linux/mfd/arizona/pdata.h
index 68890f3..23ecc31 100644
--- a/include/linux/mfd/arizona/pdata.h
+++ b/include/linux/mfd/arizona/pdata.h
@@ -150,7 +150,7 @@ struct arizona_pdata {
 	int hpdet_id_gpio;
 
 	/** Callback notifying HPDET result */
-	void (*hpdet_cb)(unsigned int measurement);
+	void (*hpdet_cb)(unsigned int measurement, bool mic);
 
 	/** Use HPDETL to check for moisture, this value specifies the
 	 * threshold impedance in ohms above which it will be considered
diff --git a/sound/soc/codecs/arizona.c b/sound/soc/codecs/arizona.c
index 8df24e3..3f2b89e 100644
--- a/sound/soc/codecs/arizona.c
+++ b/sound/soc/codecs/arizona.c
@@ -2281,7 +2281,7 @@ int arizona_set_output_mode(struct snd_soc_codec *codec, int output, bool diff)
 EXPORT_SYMBOL_GPL(arizona_set_output_mode);
 
 int arizona_set_hpdet_cb(struct snd_soc_codec *codec,
-			 void (*hpdet_cb)(unsigned int measurement))
+			 void (*hpdet_cb)(unsigned int measurement, bool mic))
 {
 	struct arizona *arizona = dev_get_drvdata(codec->dev->parent);
 
diff --git a/sound/soc/codecs/arizona.h b/sound/soc/codecs/arizona.h
index 8b23595..b17f3d6 100644
--- a/sound/soc/codecs/arizona.h
+++ b/sound/soc/codecs/arizona.h
@@ -261,7 +261,8 @@ int arizona_set_output_mode(struct snd_soc_codec *codec, int output,
 			    bool diff);
 
 extern int arizona_set_hpdet_cb(struct snd_soc_codec *codec,
-				void (*hpdet_cb)(unsigned int measurement));
+				void (*hpdet_cb)(unsigned int measurement,
+						 bool mic));
 extern int arizona_set_ez2ctrl_cb(struct snd_soc_codec *codec,
 				  void (*ez2ctrl_trigger)(void));
 extern int arizona_set_custom_jd(struct snd_soc_codec *codec,
-- 
1.7.9.5

